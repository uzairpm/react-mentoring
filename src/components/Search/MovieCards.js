import React from 'react';

import MovieCard from '../Movie/MovieCard';
import SearchFilter from './SearchFilter';
import Footer from '../Footer/Footer';

const MovieCards = (props) => {
    return (
        <div>
            <SearchFilter resultCount={props.movies.length}
                releaseDate={props.releaseDate}
                releaseDateClick={props.releaseDateClick}
                ratingClick={props.ratingClick} />
            <div className="row">
                {
                    props.movies.map(movie => {
                        return <MovieCard key={movie.id} movie={movie} />
                    })
                }
            </div>
            <Footer />
        </div>
    );
};
export default MovieCards;
