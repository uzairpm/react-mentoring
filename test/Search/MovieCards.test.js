import { shallow } from 'enzyme';
import React from 'react';

import MovieCards from '../../src/components/Search/MovieCards';

describe('MovieCards Component', () => {
    const movies = [{id: 1}, {id: 2}];
    const releaseDateClick = jest.fn();
    const ratingClick = jest.fn();
    const element = shallow(<MovieCards movies={movies}
            releaseDateClick={releaseDateClick} ratingClick={ratingClick}/>);
    test('Snapshot testing with custom values for props', () => {
        expect(element).toMatchSnapshot();
    });
});
