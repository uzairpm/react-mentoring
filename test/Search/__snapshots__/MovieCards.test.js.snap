// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component creation MovieCards component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCards
    movies={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ]
    }
    ratingClick={[MockFunction]}
    releaseDateClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchFilter
          ratingClick={[MockFunction]}
          releaseDate={undefined}
          releaseDateClick={[MockFunction]}
          resultCount={2}
        />,
        <div
          className="row"
        >
          <MovieCard
            movie={
              Object {
                "id": 1,
              }
            }
          />
          <MovieCard
            movie={
              Object {
                "id": 2,
              }
            }
          />
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ratingClick": [MockFunction],
          "releaseDate": undefined,
          "releaseDateClick": [MockFunction],
          "resultCount": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <MovieCard
              movie={
                Object {
                  "id": 1,
                }
              }
            />,
            <MovieCard
              movie={
                Object {
                  "id": 2,
                }
              }
            />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "movie": Object {
                "id": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "movie": Object {
                "id": 2,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchFilter
            ratingClick={[MockFunction]}
            releaseDate={undefined}
            releaseDateClick={[MockFunction]}
            resultCount={2}
          />,
          <div
            className="row"
          >
            <MovieCard
              movie={
                Object {
                  "id": 1,
                }
              }
            />
            <MovieCard
              movie={
                Object {
                  "id": 2,
                }
              }
            />
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ratingClick": [MockFunction],
            "releaseDate": undefined,
            "releaseDateClick": [MockFunction],
            "resultCount": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <MovieCard
                movie={
                  Object {
                    "id": 1,
                  }
                }
              />,
              <MovieCard
                movie={
                  Object {
                    "id": 2,
                  }
                }
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "movie": Object {
                  "id": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "movie": Object {
                  "id": 2,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
